tensorflow/core/kernels/gpu_prim.h

```diff
@@ -31,17 +31,27 @@ limitations under the license, the license you must see.
 #include "third_party/gpus/cuda/include/cusparse.h"
 
 namespace gpuprim = ::cub;
+
+// Required for sorting Eigen::half
+namespace cub {
+template <>
+struct NumericTraits<Eigen::half>
+    : BaseTraits<FLOATING_POINT, true, false, unsigned short, Eigen::half> {};
```

Comment:
Use int16/int64/etc, rather than the C type short  [runtime/int] [4].  Please fix.